Homework02/Kursovik/C_sharp_Kursovik_OOP/PathLine.cs

```diff
@@ -0,0 +1,376 @@
+        }
+        /// <summary>
+        /// Показать все файлы
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        public static List<FileInfo> GetFiles(String path)
+        {
+            if (path == "" || path == null)
+            {
+                return null;
+            }
+            if (Path.HasExtension(path))
+            {
+                path = Path.GetDirectoryName(path);
+            }
+            return new DirectoryInfo(path).GetFiles().ToList();
+        }
+        /// <summary>
+        /// Показать все папки и файлы в заданной директории
+        /// </summary>
+        /// <param name="dirName"></param>
+        public static void GetShow(string dirName)
+        {
+            Console.WriteLine();
+            if (Directory.Exists(dirName))
+            {
+                Console.WriteLine("Подкаталоги:");
+                string[] dirs = Directory.GetDirectories(dirName);
+                foreach (string s in dirs)
+                {
+                    Console.WriteLine(s);
+                }
+                Console.WriteLine();
+                Console.WriteLine("Файлы:");
+                string[] files = Directory.GetFiles(dirName);
+                foreach (string s in files)
+                {
+                    Console.WriteLine(s);
+                }
+            }
+        }
+        /// <summary>
+        /// Название директории
+        /// </summary>
+        /// <param name="dirName"></param>
+        /// <returns></returns>
+        public static string GetPath(string dirName)
+        {
+            if (Directory.Exists(dirName))
+            {
+                return dirName;
+            }
+            else
+                return null;
+        }
+        /// <summary>
+        /// Создать папку
+        /// </summary>
+        /// <param name="oldPath"></param>
+        /// <param name="newCatalog"></param>
+        /// <returns></returns>
+        public static string CreateCatalog(string oldPath, string newCatalog)
+        {
+            string oldPath2 = @$"{oldPath}\{newCatalog}";
+            DirectoryInfo dirInfo2 = new DirectoryInfo(oldPath2);
+            DirectoryInfo dirInfo = new DirectoryInfo(oldPath);
+            if (dirInfo2.Exists)
+            {
+                //dirInfo.Create();
+                Console.WriteLine("такой каталог уже есть");
+            }
+            else
+            {
+                dirInfo.CreateSubdirectory(newCatalog);
+                Console.WriteLine("папка успешно создана");
+            }
+            return dirInfo.FullName;
+        }
+        /// <summary>
+        /// Создать файл
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="file"></param>
+        public static void CreateFile(string path, string file)
+        {
+            using (FileStream fstream = new FileStream($@"{path}\{file}", FileMode.OpenOrCreate))
```

Comment:
```suggestion
            using (FileStream fstream = new FileStream(Path.Combine(path, file), FileMode.OpenOrCreate))
```
`Path.Combine` - безопаснее